BalancedBinaryTree.java

```diff
@@ -0,0 +1,42 @@
+// Time Complexity : O(n) where n is number of node in the binary tree
+// Space Complexity : O(h) where h is height of binary tree. in worst case the height of binary tree can be n, so space can be O(n) in that case
+// Did this code successfully run on Leetcode : yes
+// Any problem you faced while coding this : no
+
```

Comment:
Try to minimize by using one function, try not to call height for all the nodes!!